src/test/java/org/opensearch/security/dlic/dlsfls/RenameFieldResponseProcessorTest.java

```diff
@@ -0,0 +1,243 @@
+    }
+
+    @Test
+    public void testDocumentLevelSecurity() throws Exception {
+        setup(
+            new DynamicSecurityConfig().setSecurityRoles(ROLES_FILE).setSecurityRolesMapping(ROLES_MAPPINGS_FILE),
+            ClusterConfiguration.SEARCH_PIPELINE
+        );
+
+        HttpResponse res;
+        String testDocumentLevelSecurityPipeline = "{"
+            + "\"description\": \"A pipeline to rename a DLS restricted field name to a new field name\","
+            + "\"response_processors\": ["
+            + "{"
+            + "\"rename_field\": {"
+            + "\"target_field\": \"FlightDelayNew\","
+            + "\"field\": \"FlightDelay\""
+            + "}"
+            + "}"
+            + "]"
+            + "}";
+        res = rh.executePutRequest("/_search/pipeline/test-document-level-security", testDocumentLevelSecurityPipeline, asAdmin);
+        assertThat(res.getStatusCode(), equalTo(HttpStatus.SC_OK));
+
+        // Search with this pipeline should succeed and should only return documents where "FlightDelay" is true and the new
+        // "FlightDelayNew" will also be true
+        res = rh.executePostRequest("/flights/_search?search_pipeline=test-document-level-security&size=2", emptyQuery, asUserA);
+        assertThat(res.getStatusCode(), equalTo(HttpStatus.SC_OK));
+        assertThat(res.findValueInJson("hits.hits[0]._source.FlightDelayNew"), equalTo("true"));
+    }
+
+    @Test
+    public void testDocumentLevelSecurityReverse() throws Exception {
+        setup(
+            new DynamicSecurityConfig().setSecurityRoles(ROLES_FILE).setSecurityRolesMapping(ROLES_MAPPINGS_FILE),
+            ClusterConfiguration.SEARCH_PIPELINE
+        );
+
+        HttpResponse res;
+        String testDocumentLevelSecurityReversePipeline = "{"
+            + "\"description\": \"A pipeline to rename an accessible field name to a DLS restricted field name\","
+            + "\"response_processors\": ["
+            + "{"
+            + "\"rename_field\": {"
+            + "\"target_field\": \"FlightDelay\","
+            + "\"field\": \"Canceled\""
+            + "}"
+            + "}"
+            + "]"
+            + "}";
+        res = rh.executePutRequest(
+            "/_search/pipeline/test-document-level-security-reverse",
+            testDocumentLevelSecurityReversePipeline,
+            asAdmin
+        );
+        assertThat(res.getStatusCode(), equalTo(HttpStatus.SC_OK));
+
+        // Search with this pipeline should succeed and should only return documents where "FlightDelay" is true in the original document
+        // and the new "FlightDelay"
+        // will also be true
+        res = rh.executePostRequest("/flights/_search?search_pipeline=test-document-level-security-reverse&size=2", emptyQuery, asUserA);
+        assertThat(res.getStatusCode(), equalTo(HttpStatus.SC_OK));
+        assertThat(res.findValueInJson("hits.hits[0]._source.FlightDelay"), equalTo("true"));
```

Comment:
Because the first document, id =1, has `FlightDelay: true`, so in the worst case that the document security breaks, it will also assert the first hit to be `FlightDelay: true`. 

To verify the document security is working properly, should add assert the length of `hits` to be 1. There are 3  documents in total, you want to make sure it's only returning the only one document that is compiling to document level security setting.

Reply:
Nice catch! I added assertions for the DLS tests to ensure that documents that are restricted do not show up in the search results (only one out of three documents should be available to be seen, the other two are restricted by DLS).